File: java-workshop/part-1-powermock/src/main/java/infrastructure/ShippingApi.java
package infrastructure;

import domain.ShippingConfirmation;
import domain.ShippingRequest;
import java.util.UUID;

/**
 * External API client - instantiated directly in services (bad practice)
 * Requires PowerMock to mock constructor.
 */
public class ShippingApi {
    private final String baseUrl = "https://shipping.example.com";

    public ShippingConfirmation createShipment(ShippingRequest request) {
        System.out.println("Calling shipping API at " + baseUrl);
        return new ShippingConfirmation(
                request.orderId(),
                "TRACK-" + UUID.randomUUID().toString().substring(0, 8),
                "2024-12-25"
        );
    }

    public boolean cancelShipment(String trackingNumber) {
        System.out.println("Cancelling shipment via " + baseUrl + " for " + trackingNumber);
        return true;
    }
}

File: csharp-workshop/Part1.StaticDeps/Infrastructure/ShippingApi.cs
using Part1.StaticDeps.Domain;

namespace Part1.StaticDeps.Infrastructure;

/// <summary>
/// External API client — instantiated directly in services (bad practice).
/// Requires constructor mocking (JustMock/Typemock) to test in isolation.
/// Java equivalent requires PowerMock to mock the constructor.
/// </summary>
public class ShippingApi
{
    private readonly string _baseUrl = "https://shipping.example.com";

    public ShippingConfirmation CreateShipment(ShippingRequest request)
    {
        Console.WriteLine($"Calling shipping API at {_baseUrl}");
        return new ShippingConfirmation(
            request.OrderId,
            "TRACK-" + Guid.NewGuid().ToString()[..8],
            "2024-12-25");
    }

    public bool CancelShipment(string trackingNumber)
    {
        Console.WriteLine($"Cancelling shipment via {_baseUrl} for {trackingNumber}");
        return true;
    }
}
